service/src/main/java/bio/terra/axonserver/app/controller/ControllerBase.java

```diff
@@ -0,0 +1,30 @@
+package bio.terra.axonserver.app.controller;
+
+import bio.terra.axonserver.app.configuration.SamConfiguration;
+import bio.terra.common.iam.SamUser;
+import bio.terra.common.iam.SamUserFactory;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Super class for controllers containing common code. The code in here requires the @Autowired
+ * beans from the @Controller classes, so it is better as a superclass rather than static methods.
```

Comment:
IMO inheritance for code sharing is not a good use of OOP. In this case you should be able to use composition instead by making this object a `@Component` and injecting it into your controllers.

If you only need a bearer token (which appears to be the only thing used here) you could use `BearerToken` instead of `SamUser` and avoid the roundtrip to SAM (assuming APIs you use eventually check for TOS by calling SAM).

In either event, `HttpServletRequest` is a request scoped bean so it can be injected anywhere in your code, not just in the controller. If your services are only called in a request scope context (e.g. never in a thread or via a stairway flight) you can inject them directly into the component that needs it, instead of in the controller and then passing it down. An example of this: https://github.com/DataBiosphere/terra-data-catalog/blob/main/common/src/main/java/bio/terra/catalog/config/BeanConfig.java#L36

Reply:
I agree with your first point. Right, I don't call SAM directly but the api/services I use do. I can switch to just the token. Interesting tip on request scoped beans!

Reply:
In WSM we have the practice of doing access control checks in the controller layer, so we always handle the request there.

Reply:
> In WSM we have the practice of doing access control checks in the controller layer, so we always handle the request there.

I don't know about WSM, but in this case there is no access control check in the controller layer. If all the code is doing is getting the token from the http request header, that's not an access check and could be done in the service that uses the token. Getting it in the controller and passing it down via APIs doesn't add a benefit, unless your code is multithreaded, since request scope objects aren't present in threads by default.

Reply:
Added a change that calls sam to fetch the sam user's petSA token before calling gcs. That still only requires the user's access token. Does that count as an access control check? Should I go ahead and just use BearerToken and also make use of injecting `HttpServletRequest` wherever is needed instead of passing it down from controller?